modules/packages/Python.chpl

```diff
@@ -2816,6 +2876,157 @@ module Python {
+        PyList_SetSlice(this.getPyObject(), bounds.low.safeCast(Py_ssize_t),
+                        bounds.high.safeCast(Py_ssize_t) + 1,
+                        interpreter.toPythonInner(items));
+
+      } else if (!bounds.hasLowBound() && bounds.hasHighBound()) {
+        PyList_SetSlice(this.getPyObject(), min(Py_ssize_t),
+                        bounds.high.safeCast(Py_ssize_t) + 1,
+                        interpreter.toPythonInner(items));
+
+      } else if (bounds.hasLowBound() && !bounds.hasHighBound()) {
+        PyList_SetSlice(this.getPyObject(), bounds.low.safeCast(Py_ssize_t),
+                        max(Py_ssize_t), interpreter.toPythonInner(items));
+
+      } else {
+        PyList_SetSlice(this.getPyObject(), min(Py_ssize_t),
+                        max(Py_ssize_t), interpreter.toPythonInner(items));
+      }
+      this.interpreter.checkException();
+    }
+
+    /*
+      Insert an item into the list at the specified index.  Equivalent to
+      calling ``obj.insert(index, item)`` in Python.
+
+      :arg idx: The index to insert the item at
+      :arg item: The item to append
+    */
+    proc insert(idx: int, item: ?) throws {
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      PyList_Insert(this.getPyObject(), idx.safeCast(Py_ssize_t),
+                    interpreter.toPythonInner(item));
+      this.interpreter.checkException();
+    }
+
+    /*
+      Append an item to the end of the list.  Equivalent to calling
+      ``obj.append(item)`` in Python.
+
+      :arg item: The item to append
+    */
+    proc append(item: ?) throws {
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      PyList_Append(this.getPyObject(), interpreter.toPythonInner(item));
+      this.interpreter.checkException();
+    }
+
+    /*
+      Extend the list with the contents of `iterable`.  Equivalent to calling
+      ``obj.extend(iterable)`` in Python.
+
+      .. warning::
+
+         This method is only support for Python 3.13 or later!
+
+      :arg iterable: something that can be iterated over to extend the list
+    */
+    proc extend(iterable: ?) throws {
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      PyList_Extend(this.getPyObject(), interpreter.toPythonInner(iterable));
+      this.interpreter.checkException();
+    }
+
+    /*
+      Remove item at index from the list.  Equivalent to calling `del obj[idx]`
+      in Python.
+
+      :arg idx: The index of the item to remove.
+    */
+    proc remove(idx: int) throws {
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      PyList_SetSlice(this.getPyObject(), idx.safeCast(Py_ssize_t),
+                      idx.safeCast(Py_ssize_t) + 1, nil);
+      this.interpreter.checkException();
+    }
+
+    /*
+      Remove the specified items from the list.  Equivalent to calling
+      ``obj[bounds.low:bounds.high+1] = []`` in Python.
```

Comment:
I would have expected this python snippet to just reset 1 element of the range to be the literal empty list, and that the right Python code to be `del obj[bounds.low:bounds.high+1]`. Did you test this?

Reply:
I did, and I believe the delete version also works.  But I'll double check since it's out of my terminal history

Reply:
Huh. yeah if you can put both in the docs that'd be great!